instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/builder/internal/DefaultHttpServerInstrumenterBuilder.java

```diff
@@ -195,8 +194,12 @@ private InstrumenterBuilder<REQUEST, RESPONSE> builder() {
           .addAttributesExtractor(HttpExperimentalAttributesExtractor.create(attributesGetter))
           .addOperationMetrics(HttpServerExperimentalMetrics.get());
     }
+    builderCustomizer.accept(builder);
 
-    return builder;
+    if (headerGetter != null) {
+      return builder.buildServerInstrumenter(headerGetter);
+    }
+    return builder.buildInstrumenter(SpanKindExtractor.alwaysServer());
```

Comment:
are there any server instrumentations that don't provide a `headerGetter`?

Reply:
no, they all do

Reply:
made it part of the ctor

Reply:
just found that ktor doesn't have the header getter - but this is an upcoming pr